src/tools/compiletest/src/runtest.rs

```diff
@@ -2029,7 +2028,9 @@ impl<'test> TestCx<'test> {
             }
         }
 
-        if let Some(PassMode::Check) = self.pass_mode() {
+        if let (Some(PassMode::Check), ..) | (_, Some(FailMode::Check), Ui)
+            = (self.pass_mode(), self.props.fail_mode, self.config.mode)
+        {
             rustc.args(&["--emit", "metadata"]);
```

Comment:
Apparently the `!is_rustdoc` condition should be here as well, because `rustdoc` doesn't understand `--emit metadata` - https://github.com/rust-lang/rust/issues/67128.